src/zeek.bif

```diff
@@ -1185,6 +1186,58 @@ function clear_table%(v: any%): any
 	return nullptr;
 	%}
 
+## Gets all values from a table.
+##
+## t: The :zeek:type:`table`
+##
+## Returns: A ``vector of T`` of all the values in t.
+##
+## .. zeek:see:: table_keys
+function table_values%(t: any%): any_vec
+	%{
+	if ( ! t->GetType()->IsTable() )
+		{
+		zeek::emit_builtin_error("table_values() requires a table argument");
+		return nullptr;
+		}
+
+	auto vt = zeek::make_intrusive<zeek::VectorType>(t->GetType()->AsTableType()->Yield());
+	auto vv = zeek::make_intrusive<zeek::VectorVal>(std::move(vt));
+	for ( const auto& iter : *t->AsTable() )
+		{
+		vv->Append(iter.value->GetVal());
+		}
+
+	return vv;
+	%}
+
+## Gets all keys from a table.
+##
+## t: The :zeek:type:`table`
+##
+## Returns: A ``set of T`` of all the keys in t.
```

Comment:
Is there a requirement or something preventing us from using a vector here as well? While keys don't have a deterministic order currently, in combination with #2208 the returned vector would have a defined order.

Returning a vector might be more efficient as well it skips building up a new Dict and all the underlying composite hash / index structures. And at the same time, it seems the operations against the returned set of `table_keys()` (`| |`, `in`, `!in`, ?) could be done directly against the table?

@ynadji - your tickets asked for `table_keys()` to return `set[T]` - was there a specific motivation?

Reply:
> Is there a requirement or something preventing us from using a vector here as well? 

I think that if tables/sets with multiple indices, using `vector` will make the situation worse. 

- if returning a `vector of list`. The `list` type does not seem to be allowed to be used in Zeek scripts.
- if returning a `vector of vector of any`. I'm not sure whether the `vector` is allowed to store different types of data, But if I am the user of this function, I will be confused.

Reply:
Thanks. I guess vector elements can't have "index type".

Taking a step back, I'm still wondering what operations/use-cases the `set` returned by `table_keys()` allows that the table itself does not. But there are utilities/bifs taking a `set` and that `set` may now (easily) be the keys of a table.

Reply:
Yeah, it's mostly a helper/syntactic sugar. My immediate use-case was precisely what you're describing: I had a function that took a `set` and the `set` I wanted was of the keys of a `table`. Any use case that uses a `table` for data aggregation/bookkeeping but at the end of the day only needs the keys or the values is likely to use these functions. Furthermore, this is provided by a lot of languages that provide and associated array data structure, so programmers new to Zeek are likely to expect it to be present.